src/Aspire.Hosting.Azure.WebPubSub/AzureWebPubSubExtensions.cs

```diff
@@ -64,4 +94,78 @@ public static IResourceBuilder<AzureWebPubSubResource> AddAzureWebPubSub(this ID
+    {
+        var hubResource = new AzureWebPubSubHubResource(hubName, builder.Resource);
+        var hubBuilder = builder.ApplicationBuilder.CreateResourceBuilder(hubResource);
+        builder.Resource.Hubs[hubName] = hubBuilder;
+        return hubBuilder;
+    }
+
+    /// <summary>
+    /// Add event handler setting with expression
+    /// </summary>
+    /// <param name="builder">The builder for a Web PubSub hub.</param>
+    /// <param name="urlTemplateExpression">The expression to evaluate the URL template configured for the event handler.</param>
+    /// <param name="userEventPattern">The user event pattern for the event handler.</param>
+    /// <param name="systemEvents">The system events for the event handler.</param>
+    /// <param name="authSettings">The auth settings configured for the event handler.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<AzureWebPubSubHubResource> AddEventHandler(this IResourceBuilder<AzureWebPubSubHubResource> builder,
+        ReferenceExpression.ExpressionInterpolatedStringHandler urlTemplateExpression, string userEventPattern = "*", string[]? systemEvents = null, UpstreamAuthSettings? authSettings = null)
+    {
+        var urlExpression = ReferenceExpression.Create(urlTemplateExpression);
+
+        builder.Resource.EventHandlers.Add((builder, construct, hub) =>
+        {
+            var hubName = hub.Name.Value;
+
+            var hubResource = hub.Properties.Value!;
+            if (urlExpression.ManifestExpressions.Count == 0)
+            {
+                var eventHandler = GetWebPubSubEventHandler(urlExpression.Format, userEventPattern, systemEvents, authSettings);
+
+                // when urlExpression is literal string, simply add
+                hubResource.EventHandlers.Add(eventHandler);
+            }
+            else
+            {
+                var count = hubResource.EventHandlers.Count;
+                var urlParameter = new ProvisioningParameter($"{hubName}_url_{count}", typeof(string));
+                var eventHandler = GetWebPubSubEventHandler(urlParameter, userEventPattern, systemEvents, authSettings);
+                hubResource.EventHandlers.Add(eventHandler);
+                construct.Add(urlParameter);
+
+                builder.WithParameter(urlParameter.IdentifierName, urlExpression);
+            }
+        });
+        return builder;
+    }
+
+    private static WebPubSubEventHandler GetWebPubSubEventHandler(BicepValue<string> urlValue, string userEventPattern, string[]? systemEvents, UpstreamAuthSettings? authSettings)
+    {
+        var handler = new WebPubSubEventHandler
+        {
+            UrlTemplate = urlValue,
+            UserEventPattern = userEventPattern,
+        };
+
+        if (systemEvents != null)
+        {
+            handler.SystemEvents = new BicepList<string>(systemEvents.Select(i => new BicepValue<string>(i)).ToArray());
```

Comment:
I think you can do this:

```suggestion
            handler.SystemEvents = [..systemEvents];
```